### PR TITLE
Scrabble Exercise Miguel Silva

### PR DESCRIPTION
**Pull request by MiguelSilva1997**

### CODE REVIEW

File: test/scrabble_test.rb [hunk 1]
```diff
@@ -6,8 +6,26 @@
...
+
+  def test_score_with_multipliers
+    assert_equal 9, Scrabble.new.score_with_multipliers('hello', [1,2,1,1,1])
+    assert_equal 18, Scrabble.new.score_with_multipliers('hello', [1,2,1,1,1], 2)
+  end
+
+  def test_if_word_has_7_or_more_letters
+    assert_equal 58, Scrabble.new.score_with_multipliers('sparkle', [1,2,1,3,1,2,1], 2)
+    assert_equal 48, Scrabble.new.score_with_multipliers('sparky', [1,2,1,3,1,2,1], 2)
+  end
```
Comment (VictoriaVasys) [inline test/scrabble_test.rb:28]: Would have liked to see more tests for edge cases.

File: test/scrabble_test.rb [hunk 2]
```diff
@@ -6,8 +6,26 @@
...
   def test_it_can_score_a_single_letter
-    skip
+    # skip
     assert_equal 1, Scrabble.new.score("a")
     assert_equal 4, Scrabble.new.score("f")
   end
+
+  def test_it_can_score_a_word
+    assert_equal 8, Scrabble.new.score("hello")
+    assert_equal 0, Scrabble.new.score(nil)
```
Comment (VictoriaVasys) [inline test/scrabble_test.rb:16]: Since nil is not a word, it doesn't belong in this test

File: test/scrabble_test.rb [hunk 3]
```diff
@@ -6,8 +6,26 @@
 
 class ScrabbleTest < Minitest::Test
   def test_it_can_score_a_single_letter
-    skip
+    # skip
```
Comment (VictoriaVasys) [inline test/scrabble_test.rb:9]: Remove comments before pushing your final code.

File: lib/scrabble.rb [hunk 4]
```diff
@@ -1,7 +1,10 @@
 class Scrabble
 
   def score(word)
-    1
+    score = 0
+    return score if word == nil || word.empty?
+    word.each_char {|l| score += point_values[l.upcase]}
```
Comment (VictoriaVasys) [inline lib/scrabble.rb:6]: Nice use of a different enum, but don't abbreviate your params/variables ('l')

File: lib/scrabble.rb [hunk 5]
```diff
@@ -15,4 +18,13 @@ def point_values
       "Y"=>4, "Z"=>10
     }
   end
+
+  def score_with_multipliers(word, multi, doubler = 1)
+    score = 0
+    letter = word.upcase.split("")
+    letter.each_with_index {|v,i|score += (point_values[v] * multi[i]) * doubler}
```
Comment (VictoriaVasys) [inline lib/scrabble.rb:25]: This is better than using #each but push yourself to use more advanced enums (like reduce). This line is also a bit long & could benefit from breaking some logic out into different methods.

**Review by VictoriaVasys — COMMENTED** [review]
Good work overall. Push yourself to be really explicit about what each test's purpose is, test for edge cases & break your logic out.
